src/common/wayfinder/routers/random.ts

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright (C) 2022-2024 Permanent Data Solutions, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { AoARIORead } from '../../../types/io.js';
+import { WayfinderRouter } from '../../../types/wayfinder.js';
+import { randomInt } from '../wayfinder.js';
+
+export class RandomGatewayRouter implements WayfinderRouter {
+  public readonly name = 'random';
+  private ario: AoARIORead;
+  private blocklist: string[];
+  constructor({
+    ario,
```

Comment:
it may be easier to make this just anything that supports `getGateways` interface - allowing us to do robust caching of the full list of gateways